### PR TITLE
- optimizing imports and updating code style

### LINKED ISSUE #62: htsjdk and picard needed as dependencies for pipelines
Build Error:

```
[info] Compiling 49 Scala sources and 6 Java sources to .../dagr/core/target/scala-2.11/classes...
[warn] .../dagr/core/src/main/scala/dagr/core/tasksystem/Pipe.scala:162: abstract type Out1 in type pattern dagr.core.tasksystem.Pipes.EmptyPipe[Out1] is unchecked since it is eliminated by erasure
[warn]       case (p: EmptyPipe[Out1], _) => right.asInstanceOf[Pipe[In1,Out2]]
[warn]                ^
[warn] .../dagr/core/src/main/scala/dagr/core/tasksystem/Pipe.scala:163: abstract type Out2 in type pattern dagr.core.tasksystem.Pipes.EmptyPipe[Out2] is unchecked since it is eliminated by erasure
[warn]       case (_, p: EmptyPipe[Out2]) => left.asInstanceOf[Pipe[In1,Out2]]
[warn]                   ^
[warn] two warnings found
[info] Compiling 51 Scala sources to .../dagr/tasks/target/scala-2.11/classes...
[info] Compiling 6 Scala sources to .../dagr/pipelines/target/scala-2.11/classes...
[error] .../dagr/pipelines/src/main/scala/dagr/pipelines/CreateUnmappedBamFromFastqPipeline.scala:104: not found: value SortOrder
[error]       val mergeUnmappedSams = new MergeSamFiles(in=unmappedBams, out=unmappedBamFile, sortOrder=SortOrder.queryname)
[error]                                                                                                 ^
[error] .../dagr/pipelines/src/main/scala/dagr/pipelines/DownsampleAndCallSomaticVariants.scala:130: not found: type MetricsFile 
[error]     val mfile = new MetricsFile[HsMetrics,java.lang.Integer]
[error]                     ^
[error] two errors found
[error] (dagr-pipelines/compile:compileIncremental) Compilation failed
[error] Total time: 19 s, completed Feb 9, 2016 12:31:41 AM
```

We need to add htsjdk and picard as dependencies for pipelines:

```
lazy val pipelines = Project(id="dagr-pipelines", base=file("pipelines"))
  .settings(description := "A set of example dagr pipelines.")
  .settings(commonSettings: _*)
  .settings(
    libraryDependencies ++= Seq(
      "com.github.samtools"       % "htsjdk" % "2.1.0" excludeAll(htsjdkAndPicardExcludes: _*),
      "com.github.broadinstitute" % "picard" % "2.1.0" excludeAll(htsjdkAndPicardExcludes: _*)
    )
  )
  .dependsOn(tasks, core)
```

### PR DESCRIPTION
**Pull request by nh13**
- updating pipeline to depend on htsjdk and picard

The second bullet fixes #62, while the first bullet reflects 99% of the changes.

### CODE REVIEW

**Comment by nh13** [comment]
@tfenne I need to fix the tests, but #62 remains.

**Comment by nh13** [comment]
@tfenne tests are fixed. In addition to adding htsjdk and picard as dependencies in pipelines, I updated `DownsampleAndCallSomaticVariants` to get rid of the ugly `root` import.  All other changes are cosmetic.

File: core/src/main/scala/dagr/core/cmdline/parsing/ClpReflectiveBuilder.scala [hunk 1]
```diff
@@ -130,24 +134,39 @@ private[parsing] class ClpArgument(declaringClass: Class[_],
...
-  lazy val shortName: String    = annotation.map(_.flag()).getOrElse("")
-  lazy val doc: String          = annotation.map(_.doc()).getOrElse("")
-  lazy val isCommon: Boolean    = annotation.exists(_.common())
-  lazy val minElements: Int     = if (isCollection) annotation.map(_.minElements).getOrElse(1) else throw new IllegalStateException("Calling minElements on an argument that is not a collection.")
-  lazy val maxElements: Int     = if (isCollection) annotation.map(_.maxElements).getOrElse(Integer.MAX_VALUE) else throw new IllegalStateException("Calling minElements on an argument that is not a collection.")
+  lazy val longName: String = if (annotation.isDefined && annotation.get.name.nonEmpty) annotation.get.name else StringUtil.camelToGnu(name)
+  lazy val shortName: String = annotation.map(_.flag()).getOrElse("")
+  lazy val doc: String = annotation.map(_.doc()).getOrElse("")
+  lazy val isCommon: Boolean = annotation.exists(_.common())
+  lazy val minElements: Int = if (isCollection) {
```
Comment (tfenne) [inline core/src/main/scala/dagr/core/cmdline/parsing/ClpReflectiveBuilder.scala]: I don't love this layout but I'm struggling to come up with a better suggestion. Do we really need to throw an exception if this is called on a non-collection?  We could just return 0 or 1 depending on `required`...

File: core/src/main/scala/dagr/core/cmdline/parsing/DagrCommandLineParser.scala [hunk 2]
```diff
@@ -312,26 +324,27 @@ private[cmdline] class DagrCommandLineParser(val commandLineName: String, val in
...
         levenshteinDistance(command, name, 0, 2, 1, 4)
       }
       distances.put(clazz, distance)
-      if (distance < bestDistance) {
-        bestDistance = distance
-        bestN = 1
-      }
-      else if (distance == bestDistance) {
-        bestN += 1
+      distance match {
```
Comment (tfenne) [inline core/src/main/scala/dagr/core/cmdline/parsing/DagrCommandLineParser.scala]: For simple boolean checks, I find this pattern of using matches with a bunch of `case x if x ....` to be more verbose, and less clear.

File: core/src/main/scala/dagr/core/tasksystem/Pipe.scala [hunk 3]
```diff
@@ -139,8 +137,11 @@ object Pipes {
       */
     override def applyResources(resources: ResourceSet): Unit = {
       val (fixed, variable) = partition
-      var remainingResource = resources
-      fixed.foreach(task => { task.applyResources(task.resources); remainingResource -= task.resources })
+      val remainingResource = resources - fixed.map( task => {
```
Comment (tfenne) [inline core/src/main/scala/dagr/core/tasksystem/Pipe.scala]: I get that this eliminates a var, but it also goes from 2 to 5 lines of code, and I find it quite a bit less readable.

File: sopt/src/main/scala/dagr/sopt/OptionLookup.scala [hunk 4]
```diff
@@ -149,11 +149,14 @@ trait OptionLookup {
     */
   private[sopt] def findExactOrPrefix(optionName: String): List[OptionAndValues] = {
     // first see if the name is just in the map
-    if (optionMap.contains(optionName)) return List(optionMap.get(optionName).get)
-    // next, check abbreviations
-    getOptionNamesWithPrefix(optionName)
-      .map { name => optionMap.get(name).get }
-      .toList
+    if (optionMap.contains(optionName)) {
```
Comment (tfenne) [inline sopt/src/main/scala/dagr/sopt/OptionLookup.scala]: It's up to you, but I find that if each branch is a single expression, that it also looks fine without the braces.

File: core/src/main/scala/dagr/core/tasksystem/Pipe.scala (hunk 3 above)
Comment (nh13) [inline core/src/main/scala/dagr/core/tasksystem/Pipe.scala]: I will find a way to make it two lines.

**Comment by nh13** [comment]
Fixed #66

**Pull request merged**